### DISCUSSION TITLE
va-tree-view - Add children dynamically after a database call

### DISCUSSION THREAD

**Discussion by kumarimanjari** [epicmaxco/vuestic-ui#4335] [General]
Dear Team,

I am using va-tree-view component.
Is it possible to add children dynamically after a database call once a click on label happens. If yes, a working demo would be good.

Thanks & Regards
Manjari Kumari

**Comment by kumarimanjari** [comment]
Hello Team,

Could you please provide your inputs on the query?

Thanks & Regards
Manjari Kumari

**Comment by kumarimanjari** [comment]
@m0ksem - As I received responses/solutions from you in past, could you please let me know also about the question I have now.

**Reply by m0ksem** [reply]
> @m0ksem - As I received responses/solutions from you in past, could you please let me know also about the question I have now.

Hi. I think you can make something similar to [this](https://ui.vuestic.dev/play#eNqdVl1v2zYU/SsXGlA4QGwly/bixQG6LAM6rF2xBn2p8sBK1xZbSRRISkln+L/vkBL14TgdMAGJKN7vcw8vvY9e1/WqbThaR9eWy7oQlm+Siui6vvlTGktqS63MWNFOlEw7rjSb9XVcd0ofxb1m/ij5kdaVythsksi/k4jaZYlVseanWlQZZxBhyW6ZRN4cDkJM+iFVleXK9g4GDR/kvVY7xDW3UqcFw7Pc9nqrQolMVjvEk/BsWZeycv6M/IehY0pRFBDGozuDdOCCC8M3+z11XsRnLuhwuI6dNCQXzxGJx2r9BoxDQTBNqpk+Pk2qZW2xhpbSljLeiqawtHfGFeBcUxL91m0m0bmzIcqEFWtanNHmhhZeE7oO0TV9CiX02+6R2ZouYRq+fSHO7+vUSlV5t4MuIHainEXhIZvI0lwWmWbIhyjuiWN6K74CzUYzWUW5aPHOpSFjm8+dFW2VJm5Zf/OJnpNR9MiUAmUFdAjNJ9clKhWcSGBkpiH2vobl5blXQl5rsrphOkyyexjW4+4RCD+fBCFrQSnkfhIHCcKBVRZk/W8k/l+aYSfwBE4f3M7hzAvcX8k2Vxna2xe0BY/fAceF7/o5wp6NtTqoF0gfB9OJ3en0ahMV1Lclb7ySGW02m7kH92gGJpW3/GXcB4OfeQhgHDnoMtiqBq3deD6sZmkPdj79SYzOtTc8cjlk5YUzk0liw7JfBJxdT277mPdqyGMSw7C9lyWrxi66wzWJDpb/7lguagnaYl4MklkxKPUEJX4EI+aHbilG2s34EUyuJiYfGt3KVhSUK62VfsHip4nFW7ZayawVlRRH6p5bHS50dXERkA8qqrrzs9hj8yabsqbraBjVDkCDcnu9VSnqxcLRCLjNu+2/Rqa6iMHlzNkKzL0Tab7o+/KsAe81RkgFUmlVhiPWDQtEFJZEoVlk37wIF8jLVKVXr+ZtWxVc7Wzuj8LlM+mni4dwhxwx0pf2ErFOnZvDHG/3370fhU3z4XiPkyCJQj/6+eMMDnCBSyjcHEkVnUeydNfHEk1YfTGqwlXtfWGAeQHG1+A9iXCXGyvTZSPdNoa9tbVZx3Ej3TXvRKuM2zio4TzEpZAVPCeRTwI5IKY1YMRW7o4ipqqsZcH6r9oRfR4ZB0c9/uH33GTsOQebnNOvJ/a/mKcuRbTesG7HIZ1EVugd41J04rsP7/ipuyA7IX5WNAW0vyP8m40qmu4GdGq/YqYg7Ymez/aNBxCtvzd3T/j1YUJRLtHQkh7V2++UPqZ7tboaUDz8C90W3Kg=) solution.

**Reply by kumarimanjari** [reply]
Thank you @m0ksem, I will try this solution and give you the feedback.

**Reply by kumarimanjari** [reply]
Hello @m0ksem - I tried the solution but seems that I have a complicated structure, I can try to show below the structure - 
There is an array - 
```
[
        {
          id: "00",
          title: "Aeroplane",
          numberOfChildren: null,
        },
        {
          id: "50",
          title: "Ship",
          numberOfChildren: null,
        },
        {
          id: "51",
          title: "Train",
          numberOfChildren: null,
        },
        {
          id: "53",
          title: "Auto",
          numberOfChildren: null,
        },
        {
          id: "Tr",
          title: "Tram",
          numberOfChildren: null,
        },
      ],
```
They will form structure like which is by default expanded like Root folder is expanded.
- Mode of Commute 
   +Train
   +Ship
   +Aeroplane 
Once I click Train a backend call will be made based on id and this will give an array of 30 objects plus those items which doesn't have children. And it will continue to have + as folder structure when children will be there until leaf node is not there.

May be you get an idea and assist here.

**Reply by kumarimanjari** [reply]
@m0ksem - Could you please provide some hints wrt Tree structure like shown below - 
````
responseNodes: [
        {
          id: "00",
          title: "00 ",
          numberOfChildren: null,
        },
        {
          id: "50",
          title: "50",
          numberOfChildren: null,
        },
        {
          id: "51",
          title: "51",
          numberOfChildren: null,
        },
        {
          id: "53",
          title: "53",
          numberOfChildren: null,
        },
        {
          id: "AB",
          title: "AB",
          numberOfChildren: null,
        },
      ],
```

      ChildNodesFor00: [
        {
          id: "4489",
          title: "VUH",
          numberOfChildren: "3",
        },
        {
          id: "1244",
          title: "VER",
          position: "1",
          parentDocId: null,
          numberOfChildren: "3",
        },
        {
          id: "4421",
          title: "IZT",
          numberOfChildren: "0",
        },
        {
          id: "1246",
          title: "SDR",
          numberOfChildren: "0",
        },
        {
          id: "4423",
          title: "TRY",
          numberOfChildren: "0",
        },
        {
          id: "1240",
          title: "OIZ"
          numberOfChildren: "0",
        },
        {
          id: "4428",
          title: "QUI",
          numberOfChildren: "2",
        },
       
        {
          id: "442",
          title: "TZR",
          numberOfChildren: "0",
        },
        {
          id: "124",
          title: "STZ",
          numberOfChildren: "2",
        },
        {
          id: "4423",
          title: "IUZ",
          numberOfChildren: "3",
        },
        {
          id: "124",
          title: "TZR",
          numberOfChildren: "0",
        },
        {
          id: "4426",
          title: "ZTY",
          numberOfChildren: "8",
        },
        {
          id: "12406",
          title: "QWE",
          numberOfChildren: "3",
        },
        {
          id: "4423",
          title: "OUZ",
          numberOfChildren: "1",
        },
        {
          id: "1243",
          title: "PQR",
          numberOfChildren: "8",
        },
        {
          id: "4423",
          title: "GHI",
          numberOfChildren: "1",
        },
        {
          id: "1241",
          title: "ABC",
          numberOfChildren: "1",
        },
        {
          id: "1240",
          title: "XYC",
          numberOfChildren: "1",
        },
      ],`

CHILDNODESFOR00 will have more child based on numberOfChildren which will again come when API call is made.

**Reply by kumarimanjari** [reply]
@m0ksem - Any thoughts on this?

**Reply by starredev** [reply]
> > @m0ksem - As I received responses/solutions from you in past, could you please let me know also about the question I have now.
> 
> Hi. I think you can make something similar to [this](https://ui.vuestic.dev/play#eNqdVl1v2zYU/SsXGlA4QGwly/bixQG6LAM6rF2xBn2p8sBK1xZbSRRISkln+L/vkBL14TgdMAGJKN7vcw8vvY9e1/WqbThaR9eWy7oQlm+Siui6vvlTGktqS63MWNFOlEw7rjSb9XVcd0ofxb1m/ij5kdaVythsksi/k4jaZYlVseanWlQZZxBhyW6ZRN4cDkJM+iFVleXK9g4GDR/kvVY7xDW3UqcFw7Pc9nqrQolMVjvEk/BsWZeycv6M/IehY0pRFBDGozuDdOCCC8M3+z11XsRnLuhwuI6dNCQXzxGJx2r9BoxDQTBNqpk+Pk2qZW2xhpbSljLeiqawtHfGFeBcUxL91m0m0bmzIcqEFWtanNHmhhZeE7oO0TV9CiX02+6R2ZouYRq+fSHO7+vUSlV5t4MuIHainEXhIZvI0lwWmWbIhyjuiWN6K74CzUYzWUW5aPHOpSFjm8+dFW2VJm5Zf/OJnpNR9MiUAmUFdAjNJ9clKhWcSGBkpiH2vobl5blXQl5rsrphOkyyexjW4+4RCD+fBCFrQSnkfhIHCcKBVRZk/W8k/l+aYSfwBE4f3M7hzAvcX8k2Vxna2xe0BY/fAceF7/o5wp6NtTqoF0gfB9OJ3en0ahMV1Lclb7ySGW02m7kH92gGJpW3/GXcB4OfeQhgHDnoMtiqBq3deD6sZmkPdj79SYzOtTc8cjlk5YUzk0liw7JfBJxdT277mPdqyGMSw7C9lyWrxi66wzWJDpb/7lguagnaYl4MklkxKPUEJX4EI+aHbilG2s34EUyuJiYfGt3KVhSUK62VfsHip4nFW7ZayawVlRRH6p5bHS50dXERkA8qqrrzs9hj8yabsqbraBjVDkCDcnu9VSnqxcLRCLjNu+2/Rqa6iMHlzNkKzL0Tab7o+/KsAe81RkgFUmlVhiPWDQtEFJZEoVlk37wIF8jLVKVXr+ZtWxVc7Wzuj8LlM+mni4dwhxwx0pf2ErFOnZvDHG/3370fhU3z4XiPkyCJQj/6+eMMDnCBSyjcHEkVnUeydNfHEk1YfTGqwlXtfWGAeQHG1+A9iXCXGyvTZSPdNoa9tbVZx3Ej3TXvRKuM2zio4TzEpZAVPCeRTwI5IKY1YMRW7o4ipqqsZcH6r9oRfR4ZB0c9/uH33GTsOQebnNOvJ/a/mKcuRbTesG7HIZ1EVugd41J04rsP7/ipuyA7IX5WNAW0vyP8m40qmu4GdGq/YqYg7Ymez/aNBxCtvzd3T/j1YUJRLtHQkh7V2++UPqZ7tboaUDz8C90W3Kg=) solution.

Question, in my case one item can have multiple nested childrens (dynamicly). Whats the easiest way to reach that? Your example is based on fetch data on the first child

**Reply by m0ksem** [reply]
> > > @m0ksem - As I received responses/solutions from you in past, could you please let me know also about the question I have now.
> > 
> > Hi. I think you can make something similar to [this](https://ui.vuestic.dev/play#eNqdVl1v2zYU/SsXGlA4QGwly/bixQG6LAM6rF2xBn2p8sBK1xZbSRRISkln+L/vkBL14TgdMAGJKN7vcw8vvY9e1/WqbThaR9eWy7oQlm+Siui6vvlTGktqS63MWNFOlEw7rjSb9XVcd0ofxb1m/ij5kdaVythsksi/k4jaZYlVseanWlQZZxBhyW6ZRN4cDkJM+iFVleXK9g4GDR/kvVY7xDW3UqcFw7Pc9nqrQolMVjvEk/BsWZeycv6M/IehY0pRFBDGozuDdOCCC8M3+z11XsRnLuhwuI6dNCQXzxGJx2r9BoxDQTBNqpk+Pk2qZW2xhpbSljLeiqawtHfGFeBcUxL91m0m0bmzIcqEFWtanNHmhhZeE7oO0TV9CiX02+6R2ZouYRq+fSHO7+vUSlV5t4MuIHainEXhIZvI0lwWmWbIhyjuiWN6K74CzUYzWUW5aPHOpSFjm8+dFW2VJm5Zf/OJnpNR9MiUAmUFdAjNJ9clKhWcSGBkpiH2vobl5blXQl5rsrphOkyyexjW4+4RCD+fBCFrQSnkfhIHCcKBVRZk/W8k/l+aYSfwBE4f3M7hzAvcX8k2Vxna2xe0BY/fAceF7/o5wp6NtTqoF0gfB9OJ3en0ahMV1Lclb7ySGW02m7kH92gGJpW3/GXcB4OfeQhgHDnoMtiqBq3deD6sZmkPdj79SYzOtTc8cjlk5YUzk0liw7JfBJxdT277mPdqyGMSw7C9lyWrxi66wzWJDpb/7lguagnaYl4MklkxKPUEJX4EI+aHbilG2s34EUyuJiYfGt3KVhSUK62VfsHip4nFW7ZayawVlRRH6p5bHS50dXERkA8qqrrzs9hj8yabsqbraBjVDkCDcnu9VSnqxcLRCLjNu+2/Rqa6iMHlzNkKzL0Tab7o+/KsAe81RkgFUmlVhiPWDQtEFJZEoVlk37wIF8jLVKVXr+ZtWxVc7Wzuj8LlM+mni4dwhxwx0pf2ErFOnZvDHG/3370fhU3z4XiPkyCJQj/6+eMMDnCBSyjcHEkVnUeydNfHEk1YfTGqwlXtfWGAeQHG1+A9iXCXGyvTZSPdNoa9tbVZx3Ej3TXvRKuM2zio4TzEpZAVPCeRTwI5IKY1YMRW7o4ipqqsZcH6r9oRfR4ZB0c9/uH33GTsOQebnNOvJ/a/mKcuRbTesG7HIZ1EVugd41J04rsP7/ipuyA7IX5WNAW0vyP8m40qmu4GdGq/YqYg7Ymez/aNBxCtvzd3T/j1YUJRLtHQkh7V2++UPqZ7tboaUDz8C90W3Kg=) solution.
> 
> Question, in my case one item can have multiple nested childrens (dynamicly). Whats the easiest way to reach that? Your example is based on fetch data on the first child

I just made expanded option for any node available in `0.0.0-experimental-66709a0d9-20240910`. You can try it.

**Reply by starredev** [reply]
> 0.0.0-experimental-66709a0d9-20240910

Is there a example how to use that new expanded option?

**Reply by starredev** [reply]
Ah I see, v-model:expanded="expanded" in stories. WIll try!

**Reply by m0ksem** [reply]
> Ah I see, v-model:expanded="expanded" in stories. WIll try!

https://github.com/epicmaxco/vuestic-ui/blob/develop/packages/ui/src/components/va-tree-view/VaTreeView.stories.ts#L445

**Comment by kumarimanjari** [comment]
@m0ksem - We were able to create the Dynamic Navigation Tree. We are stuck at a point where using some URLs where some data is passed like http://localhost:8083/?somekey=127545811&anotherKey=00 , this should expand the node to till the leaf or let's say whatever info is passed, like tree should be expanded with these information. Is ít possible to expand the dynamic tree as well?

**Reply by kumarimanjari** [reply]
@m0ksem - Could you please assist us here

**Reply by kumarimanjari** [reply]
@m0ksem - It would be really great, if you can provide some suggestions here

**Reply by m0ksem** [reply]
> @m0ksem - It would be really great, if you can provide some suggestions here

Hi, @codeonmoon . Please, ping me tomorrow in Discord for a call. (From 14:00 to 18:00 UTC +3 Kyiv time)

**Reply by kumarimanjari** [reply]
> > @m0ksem - It would be really great, if you can provide some suggestions here
> 
> Hi, @codeonmoon . Please, ping me tomorrow in Discord for a call. (From 14:00 to 18:00 UTC +3 Kyiv time)

Sure Thank you so much
